runtime/browser/splash_screen.cc

```diff
@@ -0,0 +1,123 @@
+
+void SplashScreen::HideSplashScreen() {
+  if (!is_active_) return;
+  evas_object_hide(background_);
+  evas_object_hide(image_);
+  evas_object_del(background_);
+  evas_object_del(image_);
+  background_ = nullptr;
+  image_ = nullptr;
+  is_active_ = false;
+}
+
+std::pair<int, int> SplashScreen::GetDimensions() {
+  int w, h;
+#if defined(HAVE_X11)
+  uint16_t pid = getpid();
+  ecore_x_window_prop_property_set(elm_win_xwindow_get(window),
+                                   ECORE_X_ATOM_NET_WM_PID,
+                                   ECORE_X_ATOM_CARDINAL, 32, &pid, 1);
+  ecore_x_vsync_animator_tick_source_set(elm_win_xwindow_get(window));
+  ecore_x_window_size_get(ecore_x_window_root_first_get(), &w, &h);
+#elif defined(HAVE_WAYLAND)
+  ecore_wl_screen_size_get(&w, &h);
+#endif
+  evas_object_resize(background_, w, h);
+  return std::make_pair(w, h);
+}
+
+void SplashScreen::SetBackground(
+    const wgt::parse::SplashScreenData& splash_data, Evas_Object* parent,
+    const std::pair<int, int>& pair, const std::string& app_path) {
+  if (splash_data.background_color.empty() &&
+      splash_data.background_image.empty())
+    return;
+  background_ = elm_bg_add(parent);
+  evas_object_resize(background_, pair.first, pair.second);
+  if (!splash_data.background_image[0].empty()) {
+    elm_bg_file_set(background_,
+                    (app_path + splash_data.background_image[0]).c_str(), NULL);
+    elm_bg_option_set(background_, ELM_BG_OPTION_STRETCH);
+  } else if (!splash_data.background_color.empty()) {
+    elm_bg_color_set(background_, splash_data.color.red,
+                     splash_data.color.green, splash_data.color.blue);
+  }
+  evas_object_show(background_);
+}
+
+void SplashScreen::SetImage(const wgt::parse::SplashScreenData& splash_data,
+                            Evas_Object* parent,
+                            const std::pair<int, int>& pair,
+                            const std::string& app_path) {
+  image_ = elm_image_add(background_);
+  elm_image_file_set(image_, (app_path + splash_data.image[0]).c_str(), NULL);
+  evas_object_resize(image_, pair.first, pair.second);
+  evas_object_show(image_);
+}
+
+wgt::parse::ScreenOrientation SplashScreen::ChooseOrientation(
+    const std::map<wgt::parse::ScreenOrientation, wgt::parse::SplashScreenData>&
+        splash_map,
+    runtime::NativeWindow::ScreenOrientation screen_orientation) {
+  auto orientation_pair = splash_map.end();
+  if (screen_orientation ==
+      runtime::NativeWindow::ScreenOrientation::PORTRAIT_PRIMARY) {
+    orientation_pair = splash_map.find(wgt::parse::ScreenOrientation::PORTRAIT);
+    if (orientation_pair == splash_map.end())
+      orientation_pair = splash_map.find(wgt::parse::ScreenOrientation::AUTO);
+  } else {
+    orientation_pair =
+        splash_map.find(wgt::parse::ScreenOrientation::LANDSCAPE);
+    if (orientation_pair == splash_map.end())
+      orientation_pair = splash_map.find(wgt::parse::ScreenOrientation::AUTO);
+  }
```

Comment:
if (orientation_pair == splash_map.end())
      orientation_pair = splash_map.find(wgt::parse::ScreenOrientation::AUTO);

this code is duplicated.